lib/JSON/Validator/OpenAPI/Mojolicious.pm

```diff
@@ -191,16 +191,49 @@ sub _build_formats {
 }
 
 sub _coerce_by_collection_format {
-  my ($self, $data, $schema) = @_;
+  my ($self, $data, $p) = @_;
+
+  my $schema = $p->{schema} || $p;
   my $type = ($schema->{items} ? $schema->{items}{type} : $schema->{type}) || '';
 
-  if ($schema->{collectionFormat} eq 'multi') {
+  my $collection_format = $p->{collectionFormat};
+  my $custom_re;
+
+  # support for v3 style / explode
+  if ($p->{style}) {
+    if ($p->{style} eq 'simple') {
+      $collection_format = 'csv';
+    }
+    elsif ($p->{style} eq 'label') {
+      $custom_re = qr{\.};
+      $collection_format = $p->{explode} ? 'custom' : 'csv' if $data =~ s/^$custom_re//;
```

Comment:
Could you please elaborate when $custom_re is useful below? Could you use length() instead? Could you also send me a link to the spec where this is explained?

Reply:
Have you seen my answer given one month ago ?

Reply:
Sorry, I missed out that `$custom_re` was used outside of these if-statements.